### PR TITLE
Fail fast in release.sh when testing (#331)

### LINKED ISSUE #331: Fail fast in release.sh when testing
We don't want to wait until all unit/integration tests finish running if build tests failed, for example.

### PR DESCRIPTION
**Pull request by yt3liu**
Do not run any further tests if previous test already failed. 

Fixes #331

### CODE REVIEW

File: scripts/presubmit-tests.sh [hunk 1]
```diff
@@ -317,8 +317,14 @@ function main() {
   fi
 
   run_build_tests || failed=1
-  run_unit_tests || failed=1
-  run_integration_tests || failed=1
+  # Don't run unit tests if build tests failed
+  if (( ! failed )); then
+    run_unit_tests || failed=1
```
Comment (adrcunha) [inline scripts/presubmit-tests.sh]: You're changing the behavior or the presubmit script, not the release script. The issue was about having this behavior only for releases. I suggest using a global environment variable to make the behavior configurable.

Comment (yt3liu) [reply scripts/presubmit-tests.sh]: Updated to use the PRESUBMIT_TEST_FAIL_FAST global environment variable to configure the presubmit-test.

File: scripts/presubmit-tests.sh [hunk 2]
```diff
@@ -317,8 +317,14 @@ function main() {
   fi
```
Comment (adrcunha) [inline scripts/presubmit-tests.sh]: Also please update the README.md with the new feature.

File: test/unit/presubmit-tests.sh [hunk 3]
```diff
@@ -117,6 +128,51 @@ function test_custom_runners_basic() {
   }
 }
 
+function test_custom_runners_fail_slow() {
```
Comment (adrcunha) [inline test/unit/presubmit-tests.sh]: Great work on the unit tests, thanks!

File: scripts/presubmit-tests.sh (hunk 1 above)
Comment (yt3liu) [reply scripts/presubmit-tests.sh]: Updated to initialize the readonly env var at the beginning of the script.

File: scripts/presubmit-tests.sh [hunk 4]
```diff
@@ -317,8 +317,14 @@ function main() {
...
 
   run_build_tests || failed=1
-  run_unit_tests || failed=1
-  run_integration_tests || failed=1
+  # Don't run unit tests if build tests failed
```
[the remark this answers is not in the document]
Comment (yt3liu) [reply scripts/presubmit-tests.sh]: Updated.

File: scripts/presubmit-tests.sh (hunk 2 above)
Comment (yt3liu) [reply scripts/presubmit-tests.sh]: Updated the README to include the new configuration.

**Comment by adrcunha** [comment]
Nice PR, thanks Joyce!

/lgtm

**Pull request merged**